src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -134,35 +144,54 @@ def run(self):
             enabled = self.isEnabled(self.name)
 
             self.logger.debug("[ServiceMonitorThread] connected=%s started=%s enabled=%s", connected, started, enabled)
+            # Everithing is OK
+            if connected and started and enabled:
+                # Do we want to reset all alerts when things get back to normal?
+                ServiceMonitorThread.reset_alerts(self.name)
+                ntries = 0
+                continue
 
-            if (connected == False):
-                self.alert("attempt %d to recover service %s\n" % (ntries + 1, self.name))
+            start_service = False
+            stop_service = False
+            ntries += 1
 
-            if (connected == True) and (started == False):
-                self.logger.debug("[ServiceMonitorThread] enabling service %s", self.name)
-                try:
-                    self.middleware.call('service.start', self.name)
-                except CallTimeout:
-                    pass
+            self.alert("attempt %d to recover service %s\n" % (ntries, self.name))
+
+            if connected:
+                if not started:
+                    start_service = True
+            else:
+                if enabled:
+                    stop_service = True
 
-            elif (connected == False) and (enabled == True):
+            if stop_service:
                 self.logger.debug("[ServiceMonitorThread] disabling service %s", self.name)
                 try:
                     self.middleware.call('service.stop', self.name)
-                except CallTimeout:
+                except Exception:
+                    pass
```

Comment:
You can pass the exception in case you don't want treat it, but would be good at least log it using logger.